### DISCUSSION TITLE
Toggle Extend Layout

### DISCUSSION THREAD

**Discussion by oddacorn** [DreymaR/BigBagKbdTrixPKL#59] [Q&A] [answered]
If I want to set up the Caps Lock key to toggle the extend layer on when tapped and have that layer stay active until Caps Lock is tapped again, is that possible?

Thanks,
Todd

**Comment by DreymaR** [comment, marked as answer]
I just answered this in the Forum. Odd, heh.

> I've often considered making a state-lock (layer toggle) modifier for Extend, and you aren't the first one to ask, but so far I haven't seen enough of a point. For the normal nav/edit layer, I feel it wouldn't be an advantage as you need to move into and out of that layer all the time while using it. And my left hand pinky seems to rest comfortably on the Caps key whenever I use it for a sequence of keys.
> 
> There's also the worry that such a modifier would confuse newbs. Vim pros are familiar with moving in and out of editor states, so to them a lockable Extend state would be second nature. But other users might get worried if they got into a state-locked Extend state by accident.
> 
> Then again, some users might prefer to sequence in and out of the layer, I get that. For Linux you could quite easily make the level5 modifer a lock, and using two lv5 modifiers of the type I use in conjunction will lock/unlock the state. For Windows though, I'd have to make a new modifier type in EPKL and that'd take some coding.
> 
> So unless I really see an excellent point to it, or get sponsored to do it or something, this may not happen for Windows anytime soon.

I have added it to the TODO list, so it may happen whenever I feel like putting in the hours. But I cannot say when that may be.
